presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergColumnHandle.java

```diff
@@ -34,16 +42,27 @@
     private final ColumnIdentity columnIdentity;
     private final Type type;
     private final Optional<String> comment;
+    private final ColumnType columnType;
+    private final List<Subfield> requiredSubfields;
 
     @JsonCreator
     public IcebergColumnHandle(
             @JsonProperty("columnIdentity") ColumnIdentity columnIdentity,
             @JsonProperty("type") Type type,
-            @JsonProperty("comment") Optional<String> comment)
+            @JsonProperty("comment") Optional<String> comment,
+            @JsonProperty("columnType") ColumnType columnType,
+            @JsonProperty("requiredSubfields") List<Subfield> requiredSubfields)
```

Comment:
The `requiredSubfields` field is required because it is used later to construct the ColumnIO object.

If we don't override the `withRequiredSubfields` method, we can maintain the `requiredSubfields` object without using a list.

Reply:
Gotcha. I checked the similar implementation in the `HiveColumnHandle` and let's keep your current implementation.